src/main/java/duke/MainWindow.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import duke.commands.*;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.storage.StorageException;
+import duke.tasks.TaskList;
+import duke.ui.DialogBox;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class MainWindow extends VBox {
+    private Storage storage;
+    private TaskList taskList;
+    private Stage primaryStage;
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+    
+    private Image userImage;
+    private Image dukeImage;
+
+    /**
+     * This method is called to initialize a controller after its root element has been completely processed.
+     */
+    @FXML
+    public void initialize() {
+        userImage = new Image(getClass().getResourceAsStream("/images/user.png"));
+        dukeImage = new Image(getClass().getResourceAsStream("/images/duke.png"));
+        // Set the scroll pane to automatically scroll down when the text reaches the bottom
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+    
+    public void setComponents(Storage storage, TaskList taskList, Stage primaryStage) {
+        this.storage = storage;
+        this.taskList = taskList;
+        this.primaryStage = primaryStage;
+        loadAdditionalConstraints();
+    }
+
+    public void loadAdditionalConstraints() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        // Create a new Label control
+        String welcomeMsg = "Hello! I'm Duke.\nWhat can I do for you?";
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog(welcomeMsg, dukeImage));
+
+        primaryStage.setTitle("Duke");
+        primaryStage.setMinWidth(400.0);
+        primaryStage.setMinHeight(500.0);
+        
+    }
```

Comment:
Since this isn't a constructor, setter or getter, perhaps there should be a header comment for this method as well?

Reply:
Noted. Will add one in the next commit.